examples/step-79/doc/intro.dox

```diff
@@ -0,0 +1,597 @@
+
+    Equation 5
+@f[
+  -\int_\Omega d_{z_2}(-c_\sigma-c_{s_2}) = \int_\Omega d_{z_2} (1-\sigma-s_2)
+@f]
+
+    Equation 6
+@f[
+  -\int_\Omega   d_{y_2}(c_\rho - H(c_\sigma))=\int_\Omega d_{y_2}
+  (\rho - H(\sigma))
+@f]
+  <\li>
+  <li>Complementary Slackness - these equations essentially ensure the barrier
+  is met - in the final solution, we need $s^T z = 0$
+    Equation 7
+    @f[
+      \int_\Omega d_{s_1}(c_{s_1}z_1/s_1 +  c_{z_1} )=-\int_\Omega d_{s_1}
+      (z_1 - \alpha/s_1) ,\;\;\; \alpha \to 0
+    @f]
+
+    Equation 8
+    @f[
+      \int_\Omega d_{s_2} (c_{s_2}z_2/s_2 + c_{z_2} )=-\int_\Omega d_{s_2}
+      (z_2 - \alpha/s_2) ,\;\;\; \alpha \to 0
+    @f]
+
+    Dual Feasibility - Multiplier on slacks and slack variables must be kept
+    greater than 0. (This is the only part not implemented in the
+    "assemble\_block\_system()" function)
+    @f[
+      s,z \geq 0
+    @f]
+    <\li>
+</ol>
+
+
+
+<h3>Discretization</h3>
+We use a quadrilateral mesh with $Q_1$ element to discretize the displacement and
+ displacement Lagrange multiplier. Piecewise constant $DGQ_0$ elements are used
+ to discretize the density, unfiltered density, density slack variables, and
+ multipliers for the slack variables and filter constraint.
+
+<h3>Nonlinear Algorithm</h3>
+
+While most of the discussion above follows traditional and well-known approaches
+ to solving nonlinear optimization problems, it turns out that the problem is
+ actually quite difficult to solve in practice. In particular, it is quite
+ nonlinear and an important question is not just to find search directions
+ $c_\bullet$ as discussed above based on a Newton method, but that one needs to
+ spend quite a lot of attention to how far one wants to go in this direction.
+ This is often called ``line search" and comes down to the question of how to
+ choose the step length $\alpha_k \in (0,1]$ so that we move from the current
+ iterate $x_k$ to the next iterate $x_{k+1}=x_k+\alpha_k x_k$ in as efficient a
+ way as possible. It is well understood that we need to eventually choose
+ $\alpha_k=1$ to realize the Newton's method's quadratic convergence; however,
+ in the early iterations, taking such a long step might actually make things
+ worse, either by leading to a point that has a worse objective function or at
+ which the constraints are satisfied less well than they are at $x_k$.
+
+Very complex algorithms have been proposed to deal with this issue.
+@cite Nocedal2009 @cite Wchter2005 Here, we implement a watchdog-search
+algorithm.@cite Nocedal2006 When discussing this algorithm, we will use the
+vector $\textbf{x}$ to represent all primal variables - the filtered and
+unfiltered densities, slack variables and displacement - and use the vector
+$\textbf{y}$ to represent all of the dual vectors. The solution to the Newton
+System above will now be referred to as $\Delta \textbf{x}$ and $\Delta
+\textbf{y}$ instead of $c_{\{\cdot\}}$. A merit function is referred to here as
```

Comment:
Excuse my ignorance, but what is a merit function? My guess would be a function that somehow assigns a score to the quality of the proposed solution?

EDIT: I see that you actually introduce the concept a little later on. You might wish to mention that here.